client/src/actions/group/groupActions.js

```diff
@@ -0,0 +1,252 @@
+import axios from 'axios';
+import { browserHistory } from 'react-router';
+import actionTypes from '../actionTypes';
+
+// action for creating group
+export const createGroup = name => (dispatch) => {
+  axios.post('/api/group', { name }, { headers: { 'x-auth': window.sessionStorage.token } })
```

Comment:
Ok. First, you might want to consider creating a method that will automatically add your x-access-token to your axios request instead of adding it manually on every call. You can reach out for that.

And secondly, I don't think it's a nice idea implementing redirection/routing here. You can easily do that in your component. Although this would work, but then doing it in the component makes everything separated and makes your work neat.

You might want to consider this and make changes as applicable.